cirq/ops/raw_types.py

```diff
@@ -507,6 +507,36 @@ def validate_args(self, qubits: Sequence['cirq.Qid']):
         """
         _validate_qid_shape(self, qubits)
 
+    def _commutes_(
+        self, other: Any, *, atol: Union[int, float] = 1e-8
+    ) -> Union[bool, NotImplementedType, None]:
+        """Determine if this Operation commutes with the object"""
+        if isinstance(other, str):
```

Comment:
Why aren't we checking for `other` to be the instance of `ops.Operation`? That was the original check as well.

Reply:
1. In the prev implementation (with the strategy) we are checking that at least one of the arguments is Operation (because it might be anything when you pass to cirq.commutes(v1, v2))
2. In new implementation (a method `_commutes_` of the Operation class) we already know that `self` is an instance of the Operation class and `other` might be Operation or Gate or etc.

Reply:
The original logic was this: 

```
if not isinstance(v1, ops.Operation) or not isinstance(v2, ops.Operation):	
        return NotImplemented
```

This means that both values should be `Operation` otherwise it is not `NotImplemented`.

Reply:
Oh, sorry, you are 100% right, my bad.  Fixed in the last commit